### PR TITLE
[CELEBORN-247][FOLLOWUP] Add metrics for each user's quota usage of Celeborn Worker

### PR DESCRIPTION
**Pull request by SteNicholas**
### What changes were proposed in this pull request?

Add the metric `ResourceConsumption` to monitor each user's quota usage of Celeborn Worker.

### Why are the changes needed?

The metric `ResourceConsumption` supports to monitor each user's quota usage of Celeborn Master at present. The usage of Celeborn Worker also needs to monitor.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Internal tests.

### CODE REVIEW

**Comment by SteNicholas** [comment]
@AngersZhuuuu, could you help to review this pull request for support `ResourceConsumption` metric of worker?

**Comment by SteNicholas** [comment]
@AngersZhuuuu, could you help to review this pull request for support `ResourceConsumption` metric of Celeborn Worker?

File: worker/src/main/scala/org/apache/celeborn/service/deploy/worker/WorkerSource.scala [hunk 1]
```diff
@@ -18,13 +18,15 @@
...
 
 import org.apache.celeborn.common.CelebornConf
+import org.apache.celeborn.common.internal.Logging
 import org.apache.celeborn.common.metrics.MetricsSystem
 import org.apache.celeborn.common.metrics.source.AbstractSource
+import org.apache.celeborn.service.deploy.worker.WorkerSource._
 
-class WorkerSource(conf: CelebornConf) extends AbstractSource(conf, MetricsSystem.ROLE_WORKER) {
+class WorkerSource(conf: CelebornConf) extends AbstractSource(conf, MetricsSystem.ROLE_WORKER)
+  with Logging {
```
Comment (AngersZhuuuu) [inline worker/src/main/scala/org/apache/celeborn/service/deploy/worker/WorkerSource.scala:27]: Why with Logging? Should remove?

File: worker/src/main/scala/org/apache/celeborn/service/deploy/worker/Worker.scala [hunk 2]
```diff
@@ -509,6 +513,50 @@ private[celeborn] class Worker(
     // If worker register still failed after retry, throw exception to stop worker process
     throw new CelebornException("Register worker failed.", exception)
   }
+
+  private def handleResourceConsumption(): util.Map[UserIdentifier, ResourceConsumption] = {
+    val resourceConsumptionSnapshot = storageManager.userResourceConsumptionSnapshot()
+    resourceConsumptionSnapshot.foreach(resourceConsumption => {
```
Comment (AngersZhuuuu) [inline worker/src/main/scala/org/apache/celeborn/service/deploy/worker/Worker.scala:519]:
```
resourceConsumptionSnapshot.foreach { resourceConsumption => 
   ......
}
```

File: worker/src/main/scala/org/apache/celeborn/service/deploy/worker/Worker.scala [hunk 3]
```diff
@@ -509,6 +513,50 @@ private[celeborn] class Worker(
...
+      resourceConsumptionSource.addGauge(
+        ResourceConsumptionSource.DISK_FILE_COUNT,
+        resourceConsumption._1.toMap) { () =>
+        computeUserResourceConsumption(resourceConsumption).diskFileCount
+      }
+      resourceConsumptionSource.addGauge(
+        ResourceConsumptionSource.DISK_BYTES_WRITTEN,
+        resourceConsumption._1.toMap) { () =>
+        computeUserResourceConsumption(resourceConsumption).diskBytesWritten
+      }
```
Comment (AngersZhuuuu) [inline worker/src/main/scala/org/apache/celeborn/service/deploy/worker/Worker.scala:529]: Seems we should not register metrics here?

File: worker/src/main/scala/org/apache/celeborn/service/deploy/worker/WorkerSource.scala (hunk 1 above)
Comment (SteNicholas) [reply worker/src/main/scala/org/apache/celeborn/service/deploy/worker/WorkerSource.scala:27]: @AngersZhuuuu, the invoker of `startCleaner()` in `WorkerSource` needs with logging.

File: worker/src/main/scala/org/apache/celeborn/service/deploy/worker/Worker.scala (hunk 3 above)
Comment (SteNicholas) [reply worker/src/main/scala/org/apache/celeborn/service/deploy/worker/Worker.scala]: @AngersZhuuuu, due to invoke `addGauge[T](name: String, labels: Map[String, String] = Map.empty)(f: () => T)`, the metrics would be registered for new userIdentifier.

**Review by AngersZhuuuu — APPROVED** [review]
LGTM

File: worker/src/main/scala/org/apache/celeborn/service/deploy/worker/WorkerSource.scala (hunk 1 above)
Comment (FMX) [reply worker/src/main/scala/org/apache/celeborn/service/deploy/worker/WorkerSource.scala:27]: Actually. `startCleaner()` will use log from `Utils`. This is not necessary.

**Review by FMX — APPROVED** [review]
LGTM. Thanks.

**Pull request closed without merging**